beacon_node/beacon_chain/src/beacon_chain.rs

```diff
@@ -1358,6 +1358,14 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
             .collect::<Vec<_>>();
 
         for (i, block) in chain_segment.into_iter().enumerate() {
+            // Ensure the block is the correct structure for the fork at `block.slot()`.
+            if let Err(e) = block.fork_name(&self.spec) {
```

Comment:
This might be nice as as a `map_err(..)?`

Reply:
This function is a bit weird, it's not *actually* returning an error here so we can't use `map_err`.

Reply:
Oh of course